pkg/node/manager/manager.go

```diff
@@ -401,9 +478,40 @@ func isWindowsNode(node *v1.Node) bool {
 }
 
 // canAttachTrunk returns true if the node has capability to attach a Trunk ENI
-func canAttachTrunk(node *v1.Node) bool {
-	_, ok := node.Labels[config.HasTrunkAttachedLabel]
-	return ok
+func (m *manager) canAttachTrunk(node *v1.Node) (bool, error) {
+	if _, ok := node.Labels[config.HasTrunkAttachedLabel]; ok == true {
+		return true, nil
+	}
+	enabled, err := m.trunkEnabledInCNINode(node)
+	return enabled, err
+}
+
+func (m *manager) trunkEnabledInCNINode(node *v1.Node) (bool, error) {
+	var err error
+	if cniNode, err := m.wrapper.K8sAPI.GetCNINode(
+		types.NamespacedName{Name: node.Name},
+	); err == nil {
+		if lo.ContainsBy(cniNode.Spec.Features, func(addedFeature v1alpha1.Feature) bool {
+			return addedFeature.Name == v1alpha1.SecurityGroupsForPods
+		}) {
+			return true, err
```

Comment:
```suggestion
			return true, nil
```

Reply:
`err` must be nil in this block, but explicitly call this out may be better for reading.